internal/pkg/cli/app_deploy.go

```diff
@@ -248,15 +261,39 @@ func (opts *appDeployOpts) sourceEnvName() error {
 		opts.env = selectedEnvName
 	}
 
-	for _, envName := range envNames {
-		if opts.env == envName {
+	for _, env := range opts.projectEnvironments {
+		if opts.env == env.Name {
+			opts.targetEnvironment = env
+
 			return nil
 		}
 	}
 
 	return fmt.Errorf("invalid env name: %s", opts.env)
 }
 
+func (opts *appDeployOpts) configureClients() error {
+	defaultSessEnvRegion, err := session.DefaultWithRegion(opts.targetEnvironment.Region)
+	if err != nil {
+		return fmt.Errorf("create ECR session with region %s: %w", opts.targetEnvironment.Region, err)
+	}
+
+	// ECR client against tools account profile AND target environment region
+	opts.ecrService = ecr.New(defaultSessEnvRegion)
+
+	// app deploy CF client against tools account profile AND target environmnet region
+	opts.appDeployCfClient = cloudformation.New(defaultSessEnvRegion)
```

Comment:
This is what's deploying the actual app, right? If so, shouldn't this assume the environment manager role for that environment?

Reply:
Yeah I guess it would in the case of multi-account. It seems like it should be fine for single-account use-cases? I'll create that issue to capture storing the EnvManagerRoleARN in SSM and utilize it for app deploy.

Reply:
@kohidave For my own edification, where does that role live, in the stackset created on `archer init`?

Reply:
#361

Reply:
@somanyhs - we spin up that role when you create the environment (env init).